src/AppInstallerRepositoryCore/Rest/HttpClientHelper.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "cpprest/http_client.h"
+#include "cpprest/json.h"
+#include "HttpClientHelper.h"
+
+namespace AppInstaller::Repository::Rest
+{
+    HttpClientHelper::HttpClientHelper(const utility::string_t& url) : m_client(url), m_url(url) {}
+
+    pplx::task<web::http::http_response> HttpClientHelper::Post(const web::json::value& body)
+    {
+        AICLI_LOG(Repo, Verbose, << "Sending http POST request to: " << utility::conversions::to_utf8string(m_url));
+        web::http::http_request request{ web::http::methods::POST };
+        request.headers().set_content_type(web::http::details::mime_types::application_json);
+        request.set_body(body.serialize());
+        return MakeRequest(request);
+    }
+
+    web::json::value HttpClientHelper::HandlePost(const web::json::value& body)
+    {
+        web::json::value result;
+        HttpClientHelper::Post(body).then([&result](const web::http::http_response& response)
+            {
+                AICLI_LOG(Repo, Verbose, << "Response status: " << response.status_code());
+
+                if (response.status_code() == web::http::status_codes::OK)

```

Comment:
>if (response.status_code() == web::http::status_codes::OK) [](start = 16, length = 58)

I think we should try to capture the HTTP status and throw it outside of the wait if not OK.